webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala

```diff
@@ -1074,63 +1078,30 @@ class NodeApiService(
     }
   }
 
-  def property(req: Req, property: String, inheritedValue: Boolean) = {
+  def property(req: Req, property: String, inheritedValue: Boolean)(implicit qc: QueryContext) = {
     for {
-      optNodeIds <- req.json.flatMap(restExtractor.extractNodeIdsFromJson)
+      optNodeIds <- req.json.flatMap(restExtractor.extractNodeIdsFromJson).toIO
       nodes      <- optNodeIds match {
-                      case None          => nodeInfoService.getAll().toBox
-                      case Some(nodeIds) => nodeInfoService.getNodeInfosSeq(nodeIds).map(_.map(n => (n.id, n)).toMap).toBox
+                      case None          => nodeFactRepository.getAll()
+                      case Some(nodeIds) => nodeFactRepository.getAll().map(_.filterKeys(id => nodeIds.contains(id.value)))
                     }
-      propMap     = nodes.values.groupMapReduce(_.id)(n => n.properties.filter(_.name == property))(_ ::: _)
-
-      mapProps: Map[NodeId, List[JValue]] <-
-        if (inheritedValue) {
-          for {
-            inheritedProp <- getNodesPropertiesTree(nodes.values, List(property)).toBox
-          } yield {
 
-            import com.normation.rudder.domain.properties.JsonPropertySerialisation._
-            inheritedProp.map { case (k, v) => (k, v.map(_.toApiJsonRenderParents)) }
-          }
-        } else {
-          Full(propMap.map { case (k, v) => (k, v.map(_.toJson)) })
-        }
+      mapProps <- (if (inheritedValue) {
+                     for {
+                       inheritedProp <- getNodesPropertiesTree(nodes, List(property))
+                     } yield {
+                       import com.normation.rudder.domain.properties.JsonPropertySerialisation._
+                       inheritedProp.map { case (k, v) => (k, v.map(_.toApiJsonRenderParents)) }
+                     }
+                   } else {
+                     val propMap = nodes.values.groupMapReduce(_.id)(n => n.properties.filter(_.name == property))(_ ++ _)
+                     propMap.map { case (k, v) => (k, v.toList.map(_.toJson)) }.succeed
+                   }): IOResult[Map[NodeId, List[JValue]]]
```

Comment:
refactor: 
- `propMap` only used in the second case
- `properties` are now `Chunk` in place of `list` => `++`